sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -86,6 +86,14 @@ static void _handle_rdnss_timeout(sock_udp_ep_t *dns_server);
 #endif
 /** @} */
 
+/* Callback to hook into Prefix Information Option */
+__attribute__ ((weak))
+void gnrc_nib_rtr_adv_pio_cb(gnrc_netif_t *netif, const ndp_opt_pi_t *pio)
+{
+    (void) netif;
+    (void) pio;
+}
```

Comment:
How about this?
I moved the subnet logic to `gnrc/routing/simple_subnets/simple_subnets.c`

Reply:
Seems like a good compromise. This could be extended for other options, other subscribers might be interested in. One concern I have: how would multiple subscribers be implemented?

Reply:
> how would multiple subscribers be implemented?

I would deffer this issue until it actually comes up.
This is not a public API but internal to GNRC, let's not overcomplicate things for some hypothetical use case :wink: 

XFA might help us here if we don't want to waste precious RAM on a dynamic list.